pages/payments/index.tsx

```diff
@@ -40,17 +41,21 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
   return {
     props: {
       user,
-      userId
+      userId,
+      userProfile: user.userProfile
```

Comment:
This seems really redundant. You're sending the samething twice, first `user` then `user.userProfile` as userProfile